rust/src/controllers/invitations.rs

```diff
@@ -104,8 +109,8 @@ fn push_accepted_notification(
     let notification = build_inv_accepted_notification(conn, data);
 
     RabbitConnection::insecure_open(&get_rabbitmq_uri())
-        .and_then(|mut connection| connection.open_channel(None))
-        .and_then(|channel| {
+        .and_then(|mut connection| {
+            let channel = connection.open_channel(None)?;
             let result = send_notification(&channel, notification.to_string());
             let _ = channel.close();
             result
```

Comment:
I've refactored the way we handle the channel here so we avoid the connection being closed before the channel does